streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -473,38 +496,34 @@ public void run() {
     }
 
     /**
-     * Produces a string representation containing useful information about Kafka Streams
-     * Such as thread IDs, task IDs and a representation of the topology. This is useful
-     * in debugging scenarios.
+     * Produce a string representation containing useful information about this {@code KafkaStream} instance such as
+     * thread IDs, task IDs, and a representation of the topology DAG including {@link StateStore}s (cf.
+     * {@link TopologyBuilder} and {@link KStreamBuilder}).
+     *
      * @return A string representation of the Kafka Streams instance.
      */
     @Override
     public String toString() {
-        return toString("");
-    }
-
-    /**
-     * Produces a string representation containing useful information about Kafka Streams
-     * such as thread IDs, task IDs and a representation of the topology starting with the given indent. This is useful
-     * in debugging scenarios.
-     * @return A string representation of the Kafka Streams instance.
-     */
-    public String toString(final String indent) {
-        final StringBuilder sb = new StringBuilder(indent + "KafkaStreams processID:" + processId + "\n");
+        final StringBuilder sb = new StringBuilder("\tKafkaStreams processID:" + processId + "\n");
         for (final StreamThread thread : threads) {
-            sb.append(thread.toString(indent + "\t"));
+            sb.append(thread.toString("\t\t"));
```

Comment:
I don't think this allows the user to define their own indent level, as each nester level add `"\t"` as next indent in a hard coded way. So only the top level indent can be set by the user. That's why I removed it.

We would need two parameters for each method: `currentIndent` plus `indent` and use `indent` to modify `currentIndent` in each nesting level and just pass `indent` unmodified all the way down. If you want, I can include this change in this PR (but I think we want to have different PR for this change).